src/main/java/mekanism/common/block/BlockBasic.java

```diff
@@ -412,12 +414,27 @@ public void onBlockClicked(World world, BlockPos pos, EntityPlayer player)
 			{
 				if(!bin.bottomStack.isEmpty())
 				{
-					if(!player.isSneaking())
+					ItemStack stack;
+					if(player.isSneaking())
 					{
-						world.spawnEntity(new EntityItem(world, player.posX, player.posY, player.posZ, bin.removeStack().copy()));
+						stack = bin.remove(1).copy();
 					}
-					else {
-						world.spawnEntity(new EntityItem(world, player.posX, player.posY, player.posZ, bin.remove(1).copy()));
+					else
+					{
+						stack = bin.removeStack().copy();
+					}
+
+					if(!player.inventory.addItemStackToInventory(stack))
+					{
+						BlockPos dropPos = pos.offset(bin.facing);
```

Comment:
I think (can't test this version right now as I'm currently murdering the code) that interacting with the bin in creative breaks the bin instead of removing items from it.